spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/ExpressionResolvingHelper.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanExpressionContext;
+import org.springframework.beans.factory.config.BeanExpressionResolver;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.context.expression.StandardBeanExpressionResolver;
+import org.springframework.lang.Nullable;
+import org.springframework.util.StringUtils;
+
+/**
+ * Helper with convenient expression resolving methods.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ */
+public class ExpressionResolvingHelper implements BeanFactoryAware {
```

Comment:
wondering if it can be something like [this](https://github.com/spring-projects/spring-session/commit/ff4045acbd15dae53cac358315b58199368155c1). Maybe missing something

Reply:
Sure, looks interesting. The only thing I see is that it seems to resolve only for `String`, so we'd still need to do some null-defensive parsing for `Integer` values.

I'll try to take a look and also implement tests for this part, but maybe for this first milestone we can just leave it as is and change for the next one? Being a milestone I guess it's ok if we delete the `Helper` class later.

Reply:
Got it working, looks great, thanks!